File: easy/contains-duplicate/python/contains_duplicate.py
from typing import List
def contains_duplicate(nums: List[int]) -> bool:
    seen = set()
    for n in nums:
        if n in seen:
            return True
        seen.add(n)
    return False


if __name__ == "__main__":

    tests = [[1, 2, 3, 1], [1, 2, 3, 4], [1, 1, 1, 3, 3, 4, 3, 2, 4, 2], [], [42]]
    tests = [
        [1, 2, 3, 1],
        [1, 2, 3, 4],
        [1, 1, 1, 3, 3, 4, 3, 2, 4, 2],
        [],
        [42]
    ]
    for t in tests:
        print(contains_duplicate(t))

File: easy/contains-duplicate/cpp/contains_duplicate.cpp
#include <iostream>
#include <vector>
#include <unordered_set>
using namespace std;

bool containsDuplicate(const vector<int>& nums) {
    unordered_set<int> seen;
    
    for (int num : nums) {
        if (seen.count(num) > 0) {
            return true;
        }
        seen.insert(num);
    }
    
    return false;
}

int main() {
    // Test cases
    cout << boolalpha;
    cout << containsDuplicate({1, 2, 3, 1}) << endl; // true
    cout << containsDuplicate({1, 2, 3, 4}) << endl; // false
    cout << containsDuplicate({1, 1, 1, 3, 3, 4, 3, 2, 4, 2}) << endl; // true
    cout << containsDuplicate({}) << endl; // false
    cout << containsDuplicate({42}) << endl; // false
    
    return 0;
}
